### PR TITLE
Update to 1.12

### PR DESCRIPTION
**Pull request by AEnterprise**
update the api to 1.12, mostly fixing up params and moving assemblytable recipes to json, this needs a bit of cleaning up but if the general giste of this is fine i will clean this up and apply this to the integration table as well (i could not change a few things because they are shared)

the classes that just held some BC items and blocks have been removed as they where redunant and outdated. mods that need items from BC in code can use objectholders without having to rely on the API for compiling

### CODE REVIEW

File: api/buildcraft/api/transport/pipe/PipeApi.java [hunk 1]
```diff
@@ -135,7 +134,7 @@ public PowerTransferInfo(long transferPerTick, long lossPerTick, long resistance
     @CapabilityInject(IInjectable.class)
     private static Capability<IInjectable> capInjectable;
 
-    static {
+    public static void initCapabilites() {
```
Comment (AlexIIL) [inline api/buildcraft/api/transport/pipe/PipeApi.java]: Why?

Also, why did you remove the final modifier of all the capabilities?

Comment (AEnterprise) [reply api/buildcraft/api/transport/pipe/PipeApi.java]: final was removed because this wasn't a static block anymore, i would need to take another look if it is still needed, early in the port there where some issues with the mod lifecycle that ment these where not getting initialized in time

Comment (iTitus) [reply api/buildcraft/api/transport/pipe/PipeApi.java]: Isn't there an annotation for that?

Comment (AEnterprise) [reply api/buildcraft/api/transport/pipe/PipeApi.java]: there is (see above) but for some reason BC is doing a safety check that it actually registered (though i don't remember registering ever being able to fail, need to look into that)

**Comment by AEnterprise** [comment]
i think this is ready now
